From 5e26c4f76805482e3802aacd327d659c4565096c Mon Sep 17 00:00:00 2001
From: Chung-Hsien Hsu <chung-hsien.hsu@infineon.com>
Date: Wed, 22 Sep 2021 22:21:09 -0500
Subject: [PATCH 098/100] brcmfmac: use request_firmware_direct for loading
 board-specific nvram

Loading a board-specific nvram file is optional. If the file is not
present, we will get a warning "Direct firmware load for ... failed with
error ...", have 60 seconds delay when
CONFIG_FW_LOADER_USER_HELPER_FALLBACK is enabled, and fall back to a
common one. However, users may treat this as a type of side effect and
the file is required.

Since it is optional, use request_firmware_direct() and leave an info
message in place to avoid the delay and confusion.

Signed-off-by: Chung-Hsien Hsu <stanley.hsu@cypress.com>
Signed-off-by: Ting-Ying Li <tingying.li@infineon.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@infineon.com>
---
 .../net/wireless/broadcom/brcm80211/brcmfmac/firmware.c    | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
index d821a4758f8c..522bad6344d5 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
@@ -613,8 +613,11 @@ static int brcmf_fw_request_firmware(const struct firmware **fw,
 		strlcat(alt_path, fwctx->req->board_type, BRCMF_FW_NAME_LEN);
 		strlcat(alt_path, ".txt", BRCMF_FW_NAME_LEN);
 
-		ret = request_firmware(fw, alt_path, fwctx->dev);
-		if (ret == 0)
+		ret = request_firmware_direct(fw, alt_path, fwctx->dev);
+		if (ret)
+			brcmf_info("no board-specific nvram available (ret=%d), device will use %s\n",
+				   ret, cur->path);
+		else
 			return ret;
 	}
 
-- 
2.17.1

